src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactoryHelpers.cs

```diff
@@ -139,47 +142,59 @@ public static bool IsImmutableEnumerableType(this Type type)
             }
         }
 
+        [DynamicDependency(CreateRangeMethodNameForEnumerable, ImmutableArrayTypeName, ImmutableCollectionsNamespace)]
```

Comment:
`ImmutableCollectionsNamespace` -> The attribute takes an assembly name, not a namespace. It would be more understandable to call this `ImmutableCollectionsAssembly`.